### PR TITLE
feat: holder of drop functionality

### LINKED ISSUE #8495: HolderOf Motherboard drop
For next week we will come up with another rule where people need to be holders of Motherboard collection to claim drop
- https://canary.kodadot.xyz/ahp/collection/50

image

- Let @exezbcz prepare the design and we can make it ready as a secondary drop for next week
- ~~Drop will be free~~, yet the required balance for this drop would be 2 DOT on Assets Hub to make it more **premium**
- @vikiival can drop a snippet if needed something on frontend (?)
- https://github.com/kodadot/nft-gallery/issues/8482

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

  👇 __ Let's make a quick check before the contribution.

  ## PR Type

  - [x] Feature

  ## Needs QA check

  - @kodadot/qa-guild please review

  ## Context

  - [x] Closes #8495
  - [ ] Requires deployment <snek/rubick/worker>

  #### Did your issue had any of the "$" label on it?

  - [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=16SjUbGKSdjCdWTy3NNT3JxbRVGGqD4mwkHpc6BD9U2Rp29Z)

  ## Screenshot 📸

  - [ ] My fix has changed UI

image

  ## Copilot Summary
  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9b8bae7</samp>

This pull request implements a new feature that allows creating and claiming NFTs from drop campaigns that are exclusive to holders of a specific collection. It adds new translation keys, a new service function, a new component, a new drop page logic, a new type property, and two new GraphQL queries to support this feature.

  
### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 9b8bae7</samp>

> _To reward NFT holders with flair_
> _We added a badge they can wear_
> _`CollectionDropHolderOfGenerative`_
> _Shows the collection and holder incentive_
> _And the waifu service can mint them with care_

### CODE REVIEW

**Comment by Jarsen136** [comment]
The minting process for the holder of the drop is more complicated than I expected. I will leave some UI or modal issues to the follow-up issue. This PR would focus on the functionality of the `HolderOf drop`

**Comment by Jarsen136** [comment]
I made a test transaction but it failed. Could you pls help to check the minting process to find what's wrong? @vikiival 
https://assethub-kusama.subscan.io/extrinsic/5996332-2?tab=event

**Comment by vikiival** [comment]
This address does not own any ChaosFlakes

https://assethub-kusama.subscan.io/account/J23zaM8DDUewdGtrS8Vo7VSiTYrwaKpKdQ5qTTp5BDQNaMs


Claim your chaosflake and then the Id you have use as witnes
https://beta.kodadot.xyz/ahk/drops/chaosflakes

**Comment by prury** [comment]
lmk if i can help

**Comment by Jarsen136** [comment]
> lmk if i can help

Thanks. We are making some updates and testing it. I will change the `draft` status of this PR when it's ready to review.

**Comment by Jarsen136** [comment]
> Is there a copy of the funds requirement needed? The drop costs money no
> 
> image

Here is it https://github.com/kodadot/nft-gallery/issues/8495#issue-2033086223 [2 DOT on Assets Hub]

**Comment by vikiival** [comment]
Where does this number comes from ? 

It should be `itemDeposit` + `metaDataDeposit` + txFee 

actually same fn as for minting

Screenshot 2023-12-14 at 19 05 59

File: services/waifu.ts [hunk 1]
```diff
@@ -62,6 +62,12 @@ type DoRequest = {
   email?: string
 }
 
+type DoSubmitDropRequest = {
+  sn: string
+  hash: string
+  account: string
+}
```
Comment (vikiival) [inline services/waifu.ts:69]: email?: string

File: services/waifu.ts [hunk 2]
```diff
@@ -87,6 +93,20 @@ export const doWaifu = async (body: DoRequest, campaign: string) => {
   return value
 }
 
+export const doSubmitDrop = async (
+  body: DoSubmitDropRequest,
+  campaign: string,
+) => {
+  const value = await api<DoResponse>(`do/${campaign}`, {
```
Comment (vikiival) [inline services/waifu.ts:100]: For testing you can use `claimDropItem` but you need to supply image and construct metadata

**Review by vikiival — CHANGES_REQUESTED** [review]

File: services/waifu.ts (hunk 1 above)
Comment (Jarsen136) [reply services/waifu.ts:70]: updated

**Comment by Jarsen136** [comment]
> Where does this number comes from ?
> 
> It should be `itemDeposit` + `metaDataDeposit` + txFee
> 
> actually same fn as for minting
> 
> Screenshot 2023-12-14 at 19 05 59

Updated 
image

**Comment by exezbcz** [comment]
I guess this pr is needed in 30 minutes no? 

cc @vikiival

**Comment by vikiival** [comment]
> I guess this pr is needed in 30 minutes no?

yes 

@Jarsen136 please do not overwrite the prod stuff

**Comment by Jarsen136** [comment]
> @Jarsen136 please do not overwrite the prod stuff

Oki, I would not change anything. Let me know if there is something I could help : )

**Comment by prury** [comment]
flatwhite not showing requirement for me? 

image

**Comment by vikiival** [comment]
It was updated to prod value :)

**Comment by exezbcz** [comment]
@prury 
image

**Comment by yangwao** [comment]
Mobile?

**Comment by prury** [comment]
guess i will have to test it live, eligibility seems to be working ok

**Pull request merged**